llvm/lib/IR/Type.cpp

```diff
@@ -879,6 +879,10 @@ static TargetTypeInfo getTargetTypeInfo(const TargetExtType *Ty) {
         ScalableVectorType::get(Type::getInt8Ty(C), TotalNumElts));
   }
 
+  // DirectX intangible types
+  if (Name.starts_with("dx."))
+    return TargetTypeInfo(PointerType::get(C, 0), TargetExtType::CanBeGlobal);
```

Comment:
What does the `CanBeGlobal` flag do? I notice that `spirv.` also gets `HasZeroInit`. Are we sure we don't need that here as well?

Reply:
`CanBeGlobal` says that this type can be used for a global. It's not clear to me if we need that for it to be a member of a global (as in a global `RWBuffer`) so I'm not sure if we want it or not. @hekota do you know if we need this? If it's here just in case let's drop it for now and add it when we need it.

I'm pretty sure we don't want `HasZeroInit` - the only way we want to be able to get these types is from the handle creation intrinsics, so zero initialization for them doesn't make sense.

Reply:
I've removed `CanBeGlobal`. I copied it from the spirv.Image case, but since we always have the handle wrapped in a class we should not need it.